### ISSUE TITLE
Deezer playlists only use the first 386 elements

### ISSUE THREAD

**Issue by kalon33** [LMS-Community/slimserver#596]
When playing a playlist from Deezer, only the first 386 elements are displayed and used, and not the whole playlist.

**Comment by mherger** [comment]
Is this after importing it into your library, or when playing through the Deezer menu? Could you share the playlist with me?

**Comment by kalon33** [comment]
It seems both are concerned. It's my Loved tracks playlist, that I copied to a new playlist thinking that would help... I shared it for the time you would need it: https://deezer.page.link/WSqyMRBenWbw6GJ19

**Comment by kalon33** [comment]
@mherger : do you need some further info, logs, or something?

**Comment by michaelherger** [comment]
Could you please double-check the imported playlists? After reviewing the code I'd say it should import 2500 items max. Did you increase LMS' maximum playlist length from its default 500?

**Comment by kalon33** [comment]
@mherger Yes I doubled checked, and no I didn't increased LMS' maximum playlist length. I changed it to 2500 now, refreshed the music library, and still the playlist only has 386 elements (see screenshot)
![Capture d’écran de 2021-05-05 09-43-29](https://user-images.githubusercontent.com/460731/117110534-8c89dc00-ad86-11eb-99e4-9dc75f891dab.png)

**Comment by kalon33** [comment]
Dear @michaelherger , any news about this? I tried latest LMS git snapshot, without any improvement on this.

**Comment by mherger** [comment]
I'm sorry, I haven't found time to work on this yet. As I'll have to subscribe to Deezer before I can look into this I want to be sure I have the spare time to do so.

**Comment by kalon33** [comment]
@mherger If needed I may have some codes to share to get 3months free if that's not what you get when you subscribe

**Comment by michaelherger** [comment]
That could increase the motivation indeed 😁.

**Comment by kalon33** [comment]
@mherger @michaelherger TBH, as my family and I use it on a daily basis, I would be ready to bounty fixes for this one and for #623 (more globally what enhance or fix Deezer support in LMS). Not sure I would be able to sponsor the full developer time needed on it, but part of it sure if that helps to get fixes for these bugs.

**Comment by mherger** [comment]
I checked the call for playlists again, and we do indeed not provide a specific limit there. Checking on https://developers.deezer.com/api/explorer I would indeed only get 386 tracks back. But I can add `limit=2500` and would get the (almost) full list: I'd get 980, even though the header says there were 994 tracks. And if I use `index` to page through the results, using `387` at the starting point, I'd get some of the tracks that were in the first chunk already. This might be a problem with rights, eg. I'm not allowed to access some of those tracks because I'm living in a different country. But I assume that having 980 out of 994 is still better than getting 386 tracks only.

**Comment by mherger** [comment]
I've added the `limit` parameter to the playlist call. Could you please give it a try? (and please be patient: fetching information for 1k tracks takes a tad bit longer than before 😁 )

**Comment by kalon33** [comment]
OK, I now get 980 elements too, and got 1058 from another playlist I tested. Good news is that these numbers are perfectly equal to the ones displayed in Deezer's own Android app. But, looking at them through Deezer website, I got 994 elements, and 1072 items respectively, on the summary page from the playlist. Who's true? Should I count? ^^
   
Seriously, it now seems that it's fixed and problem's on Deezer side and not yours/ours. Maybe someone should report it to them, but LMS is working properly. BTW, the fetching time is pretty short (10s max maybe before completion?) and it even seems to play before everything is loaded, which leads to a few seconds gain :)
